### PR TITLE
feat!: enabling vulnerability and audit modes for workloads

### PR DESCRIPTION
**Pull request by mmorejon**
Enable vulnerability and audit modes for workloads.

### CODE REVIEW

File: autogen/safer-cluster/variables.tf.tmpl [hunk 1]
```diff
@@ -496,3 +496,15 @@ variable "enable_mesh_certificates" {
   default = false
   description = "Controls the issuance of workload mTLS certificates. When enabled the GKE Workload Identity Certificates controller and node agent will be deployed in the cluster. Requires Workload Identity."
 }
+
+variable "workload_vulnerability_mode" {
+  description = "(beta) Vulnerability mode."
+  type        = string
+  default     = ""
```
Comment (apeabody) [inline autogen/safer-cluster/variables.tf.tmpl:503]: Are these new default values going to cause a replacement for existing clusters? Could we use `null`?

Comment (mmorejon) [reply autogen/safer-cluster/variables.tf.tmpl:503]: `workload_config_audit_mode` cannot be `null`.  This value is required. I got this message setting the value as `null`.
```
The argument "protect_config.0.workload_config.0.audit_mode" is required, but no definition was found.
```

`workload_vulnerability_mode` has the same behavior using an empty string or null value. I got a change in the tfstate only when `DISABLED` or `BASIC` is used.
```
~ protect_config {
   ~ workload_vulnerability_mode = "WORKLOAD_VULNERABILITY_MODE_UNSPECIFIED" -> "DISABLED"
}
```
and
```
~ protect_config {
   ~ workload_vulnerability_mode = "WORKLOAD_VULNERABILITY_MODE_UNSPECIFIED" -> "BASIC"
}
```

Comment (mmorejon) [reply autogen/safer-cluster/variables.tf.tmpl:503]: @apeabody , what do you think?

**Comment by mmorejon** [comment]
Do you have any other suggestions? @apeabody

File: autogen/safer-cluster/variables.tf.tmpl (hunk 1 above)
Comment (mmorejon) [reply autogen/safer-cluster/variables.tf.tmpl:503]: Done. I set back the default value to empty.

**Review by apeabody — APPROVED** [review]
Thanks for the contribution @mmorejon!

**Pull request merged**